engine/access/collection_sync/collection_sync.go

```diff
@@ -0,0 +1,76 @@
+package collection_sync
```

Comment:
what do you think about structuring this like
* `engine/access/ingestion/blocks`
* `engine/access/ingestion/collections`
* `engine/access/ingestion/collections/executiondata`
* `engine/access/ingestion/collections/factory`
* `engine/access/ingestion/collections/fetcher`
* `engine/access/ingestion/receipts`